org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java

```diff
@@ -95,8 +99,20 @@ public void onEvent(Event event) throws Exception {
 		String eventName = event.getName();
 
 		if (eventName.equals(Events.ON_CLICK)) {
-			statusPanel.refresh();
-			statusPanel.updateUI();
+    		BusyDialog busyDialog = new BusyDialog();
+            busyDialog.setShadow(false);
+            getParent().insertBefore(busyDialog, getParent().getFirstChild());
+			ServerPushTemplate template = new ServerPushTemplate(AEnv.getDesktop());

```

Comment:
we are in event listener thread, ok to just use "getDesktop()" here

Reply:
ready